File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<vector<int>> sizes) {
    int answer = 0;
    int max_long = 0;
    int max_short = 0;
    
    for (vector<int> size : sizes) {
        int w = size[0];
        int h = size[1];
        
        int l = max(w, h);
        int s = min(w, h);
        
        max_long = max(max_long, l);
        max_short = max(max_short, s);
    }
    
    answer = max_long * max_short;
    
    return answer;
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int answer = 0;
        int maxW = -1;
        int maxH = -1;
        
        for (int[] s : sizes) {
            int w = s[0];
            int h = s[1];
            if (h > w) {
                int tmp = w;
                w = h;
                h = tmp;
            }
            if (w > maxW) {
                maxW = w;
            }
            if (h > maxH) {
                maxH = h;
            }
        }
        answer = maxW * maxH;
        
        return answer;
    }
}